frontend/src/components/common/Catalog/Catalog.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from 'react';
+import { ICONS } from '../../ui-kit/icons';
+import styles from './Catalog.module.scss';
+import mockCategories from '../../../data/catalog/catalogMockData.json';
+import {
+  mockContacts,
+  mockCategories as mockData,
+} from '../../../data/catalog/contatct';
+import { useNavigate } from 'react-router-dom';
+
+const Catalog = ({ setIsOpen }) => {
+  const navigate = useNavigate();
+  const [hoveredItem, setHoveredItem] = useState(null);
+
+  const arrIcon = mockData.map((category) => category.icon);
+  const arrBg = mockData.map((category) => category.bg);
+  const formatedCategories = mockCategories
+    .filter((category) => !category.parent)
+    .map((category, index) => {
+      const mainPath = `/catalog/${category.alias.toLocaleLowerCase()}`;
+      return {
+        ...category,
+        children: mockCategories
+          .filter((child) => child.parent?.id === category.id)
+          .map((child) => ({
+            ...child,
+            link: `${mainPath}/${child.alias.toLocaleLowerCase()}`,
+          })),
+        icon: arrIcon[index] || <ICONS.bukety_z_kvitiv />,
+        bg: arrBg[index],
+        link: mainPath,
+      };
+    });
+
+  const redirectHandler = (link) => {
+    navigate(link);
+    setIsOpen(false);
+  };
+
+  return (
+    <div className={styles.categoryWrapper}>
+      <div className={styles.itemsWrapper}>
+        <ul className={styles.categoryList}>
+          {formatedCategories.map((category) => {
+            const isHasChildren = !!category?.children?.length;
+            return (
+              <li
+                key={category.id}
+                className={styles.categoryItemWrapper}
+                onMouseOver={() =>
+                  setHoveredItem({
+                    subCategories: category.children,
+                    bg: category.bg,
+                    name: category.name,
+                  })
+                }
+              >
+                <a
```

Comment:
You can use the Link from the react router library, using the to prop and onClick function only to close the menu and you won't need to use this strange function